Deck builder redesign

DanieloV opened the pull request:
Modifies the deckbuilder to be similar to other online TCGs.
- catalogue style display of cards
- improved filtering options
- adjustable card sizes for better web experience
- improved display of deck cards
- sound effects on card added/removed

Doesn't include any changes related to the buttons on the right side. 

Hope this is something that you would be interested in adding.

<img width="1437" alt="Screenshot 2024-07-25 at 15 02 36" src="https://github.com/user-attachments/assets/b07a509e-7fd4-4723-b20d-2ea10404064d">

sindreslungaard commented:
I like this, feels very good to use!

My only feedback is that
- Some of the styles and colors doesn't match the rest of the site that well
- By default it downloads all card scans at once. This is nice but only works well with good internet. Trying to load the deck builder on a bad wifi connection (which many of our users struggle with) took more than 5 minutes to complete as it had to download more than 100MB. I think the best way to combat this would be to only display the <img> element if it is on screen and if not display a <div> or similar of equal size so that the scrollbar keeps consistent. It won't be a big problem for users once they've already cached all the scans to disk, but for new users it might be a bad experience

Maybe we could put these changes into a new component/view and add a checkbox to the settings page that lets users opt in to the new design and try it out and give some feedback before we make it the default. I'm sure many will test it if mentioned on discord

DanieloV commented:
Glad you like it!

- I knew the deck loading could be an issue but I didn't imagine it would be such a big one. I added lazy loading and a space taker to avoid the jumping around issue.
- Also added all of it as a different component, that can be enabled from settings. It still keeps the same path as the other one.
- In terms of color and styles, I don't really know what to change, but I am open for suggestions.
- Also made this dependent on https://github.com/sindreslungaard/duel-masters/pull/233 since some of the new code would need to be modified anyway for Vue3.
